FontCascadeCache: tune constants for platforms other than iOS family

vitorroriz opened the pull request:
#### ec5c28c0e698b3cd3c028f50141923a95a29dcde
<pre>
FontCascadeCache: tune constants for platforms other than iOS family
<a href="https://bugs.webkit.org/show_bug.cgi?id=289881">https://bugs.webkit.org/show_bug.cgi?id=289881</a>
<a href="https://rdar.apple.com/problem/147186814">rdar://problem/147186814</a>

Reviewed by Matthieu Dubet.

After [1] we are more likely to produce
hits for FontCascadeCache.

We are tunning the cache&apos;s constants here.

This is done just for platforms that are not from
the iOS family due to the more relaxed memory constraints.

[1]: <a href="https://commits.webkit.org/291808@main">https://commits.webkit.org/291808@main</a>

* Source/WebCore/platform/graphics/FontCascadeCache.cpp:
(WebCore::FontCascadeCache::retrieveOrAddCachedFonts):

Canonical link: <a href="https://commits.webkit.org/292258@main">https://commits.webkit.org/292258@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8437c5054ae549097d89b9f41e16ce92c5bb12c7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95416 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15019 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4884 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100465 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45922 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97463 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15306 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23450 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72775 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30040 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98418 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/11450 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86134 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53106 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11163 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3865 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45258 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81343 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3960 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102502 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22466 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16400 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81792 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22714 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82144 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81163 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20305 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25739 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3163 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15766 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22435 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27574 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22094 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25569 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23834 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebCore/platform/graphics/FontCascadeCache.cpp:
```diff
@@ -118,8 +118,14 @@ Ref<FontCascadeFonts> FontCascadeCache::retrieveOrAddCachedFonts(const FontCasca
     Ref<FontCascadeFonts> fonts = newEntry->fonts.get();
 
+
+#if !PLATFORM(IOS_FAMILY)
```
I don’t think OS variant is the best way to select this behavior. Some iPhone models have more RAM than some Mac models. Meanwhile some non-iOS set top boxes have less.

Probably the best thing to do is to pick the maximum that would ever be useful and the minimum that would ever be acceptable, and scale between the two based on RAM size.

geoffreygaren commented on Source/WebCore/platform/graphics/FontCascadeCache.cpp:
```diff
@@ -118,8 +118,14 @@ Ref<FontCascadeFonts> FontCascadeCache::retrieveOrAddCachedFonts(const FontCasca
 
+
+#if !PLATFORM(IOS_FAMILY)
+    static constexpr unsigned unreferencedPruneInterval = 1000;
```
What is the observed memory cost of this change, on workloads where it shows up? Maybe that can help us pick a scaling factor.

vitorroriz committed:
FontCascadeCache: tune constants for platforms other than iOS family
https://bugs.webkit.org/show_bug.cgi?id=289881
rdar://problem/147186814

Reviewed by Matthieu Dubet.

After [1] we are more likely to produce
hits for FontCascadeCache.

We are tunning the cache's constants here.

This is done just for platforms that are not from
the iOS family due to the more relaxed memory constraints.

[1]: https://commits.webkit.org/291808@main

* Source/WebCore/platform/graphics/FontCascadeCache.cpp:
(WebCore::FontCascadeCache::retrieveOrAddCachedFonts):

Canonical link: https://commits.webkit.org/292258@main
